src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -320,7 +363,7 @@ public Buffer rebuffer(long position)
                 Key key = new Key(source, internedPath, pageAlignedPos);
                 while (true)
                 {
-                    buf = cache.get(key).reference();
+                    buf = cache.get(key).loadBuffer().reference();
```

Comment:
If we fail to load the buffer, I think we might want to consider invalidating the cache.

Reply:
fixed and added a test case